### PR TITLE
#2514 - In Forum Discussion Reply, user post second GIF, it gives dup…

### LINKED ISSUE #2514: In Forum Discussion Reply, user post second GIF, it gives duplicate reply detected error - even the two GIFs were different
**Describe the bug**
In forum reply, if the user (non admin ) posts the second GIF, it gives duplicate reply error even the two GIF were different

function ref : bbp_check_for_duplicate

**To Reproduce**
1. Go to any discussion while logged in as a non admin
2. Post only GIF in reply
3. Similarly, post another different GIF in reply

**Expected behavior**
Different GIF can be posted

**Screenshots**
image

**Support ticket links**
If applicable, add HelpScout link or ticket number where the issue was originally reported.
https://secure.helpscout.net/conversation/1517114538/142860/

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2514 .

### How to test the changes in this Pull Request:

1. Go to any discussion while logged in as a non admin
2. Post only GIF in reply
3. Similarly, post another different GIF in reply

### Proof Screenshots or Video
https://vimeo.com/595850783/ef30045f0c

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — APPROVED** [review]
@milanhirapra Approved. Ready for testing.

**Comment by sumitbuddyboss** [comment]
Hello @milanhirapra 

Issue >
I have checked in fix/2514 branch and Getting an error while uploading gif.

1. How to reproduce.
2. Goto forum 
3. add a reply
4. go to forum page 1 if you have a pagination
5. then again add a new reply with a gif
6. see the error

for the proof and how to reproduce check the below issue video.

Issue video:
https://www.loom.com/share/f0fcf25444324db397885131464d2bd8

Thank you

**Comment by sumitbuddyboss** [comment]
Hello @milanhirapra 

**Fixed >**
I have checked in fix/2514 branch and cross check with @JetalDobariya @mohit-buddyboss it's working as expected for the proof please check the below video.

**Video:**
https://www.loom.com/share/16f3043993664902b4b25e65697cfaa9

Ready for release 🚀 

@milanhirapra 
Please merge this PR to qa-ongoing branch.

Thank you

File: src/bp-forums/common/functions.php [hunk 1]
```diff
@@ -721,6 +721,11 @@ function bbp_check_for_duplicate( $post_data = array() ) {
 		'check_for_duplicate'
 	);
 
+	// If reply/topic content is NULL then skip checking duplicate post
+	if( empty( $r['post_content'] ) ){
```
Comment (chetansatasiya) [inline src/bp-forums/common/functions.php:725]:
```suggestion
	if ( empty( $r['post_content'] ) ) {
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@milanhirapra Follow the spacing properly.

@dipskakadiya We need to make sure all the changes should follow the PHPCS properly.

**Review by KartikSuthar — APPROVED** [review]

**Comment by JetalDobariya** [comment]
Hello @milanhirapra 

I did verify this PR in qa-approved branch, now if we post 2 different GIF then it is working fine.
--> https://www.loom.com/i/f0b881eab51a4e8281dc21e7ba4a2679

Thanks!

**Pull request merged**